src/main/scala/rps/services/GameService.scala

```diff
@@ -7,21 +7,39 @@ import rps.models._
 import rps.models.Result._
 import rps.models.Move._
 
-object GameService {
-  def play(userMove: Move): (Move, Move, Result) = {
+import rps.services._
+import rps.repositories.GameRepository
+
+trait GameService {
+  def getGameResult(): Option[GameResult]
+  def playMove(move: Move): Unit
+}
+
+class GameServiceImpl(gameRepository: GameRepository) extends GameService {
+  def getGameResult(): Option[GameResult] = gameRepository.getGame
+  
+  def playMove(userMove: Move): Unit = {
     val enemyMove = getRandomMove
     val result = getResult(userMove, enemyMove)
 
-    (userMove, enemyMove, result)
+    gameRepository.saveGame(
+      GameResult.tupled(
+        (
+          userMove, 
+          enemyMove, 
+          result
+        )
+      )
```

Comment:
no need to use `.tupled` here
```suggestion
      GameResult(userMove, enemyMove,  result)
```